Makefile

```diff
@@ -484,7 +484,8 @@ set-manifest-image:
 ## --------------------------------------
 
 ## latest git tag for the commit, e.g., v0.3.10
-RELEASE_TAG := $(shell git describe --abbrev=0 2>/dev/null)
+RELEASE_TAG ?= $(shell git describe --abbrev=0 2>/dev/null)
+PREVIOUS_TAG ?= $(shell git describe --match "v[0-9].[0-9].[0-9]" --abbrev=0 $(RELEASE_TAG)^ 2>/dev/null)
```

Comment:
@vincepri I added a match pattern, I thing this is a lot easier and will make sure it doesn't include pre-releases:

```
➜  cluster-api git:(release-action) git describe --abbrev=0 v0.3.7^ 2>/dev/null
v0.3.7-rc.1
➜  cluster-api git:(release-action) git describe --match "v[0-9].[0-9].[0-9]" --abbrev=0 v0.4.0^ 2>/dev/null
v0.3.9
➜  cluster-api git:(release-action) git describe --match "v[0-9].[0-9].[0-9]" --abbrev=0 v0.3.7^ 2>/dev/null
v0.3.6
➜  cluster-api git:(release-action) git describe --match "v[0-9].[0-9].[0-9]" --abbrev=0 v0.4.2^ 2>/dev/null
v0.4.1
```

Reply:
Shouldn't the second one return v0.3.22?

Reply:
yes it should 

/hold

Reply:
okay this should be better... nothing a little regex can't fix 😄 

```
➜  cluster-api git:(release-action) git tag -l | sort -V | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | grep -B1 v0.3.22 | head -n 1
v0.3.21
➜  cluster-api git:(release-action) git tag -l | sort -V | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | grep -B1 v0.4.0 | head -n 1
v0.3.23
➜  cluster-api git:(release-action) git tag -l | sort -V | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | grep -B1 v0.3.4 | head -n 1
v0.3.3
➜  cluster-api git:(release-action) git tag -l | sort -V | grep -E "^v[0-9]+\.[0-9]+\.[0-9]+$" | grep -B1 v0.3.7 | head -n 1
v0.3.6
```

git describe doesn't sort versions, it sorts alphabetically, unfortunately.